clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2683,6 +2702,14 @@ private void rescheduleFindCoordinatorTask(ConsumerGroupOperationContext<?, ?> c
                 context.node().orElse(null));
         // Requeue the task so that we can try with new coordinator
         context.setNode(null);
+
+        Call call = nextCall.get();
+
+        if (call.tries() > maxRetries) {
```

Comment:
Is it possible to move the check to the AdminClientRunnable::enqueue to make this code pieces reusable?

Reply:
What about adding a new class method in Call?

Reply:
We could but we still have to do a check as to whether or not we failed the call to make sure we return from the method. It would still produce the same amount of code and I'm not sure if it would be any more efficient.

Reply:
And I'm not sure if we should move the check to `AdminClientRunnable::call/enqueue` because this is pretty specific to the consumer groups API. However if we see there's a potential reuse of code elsewhere we can definitely consider doing so.